### ISSUE TITLE
C-[ not recognized on Windows

### ISSUE THREAD

**Issue by superlou** [helix-editor/helix#677]
### Reproduction steps

On Windows 10 in both Powershell and CMD, C-[ does not match the behavior of Esc as it does on Linux. This was discovered during the conversation in #661. Explicitly mapping it as in the following `config.toml` does not seem to be recognized either:

```
theme = "bogster"

[keys.insert]
"C-[" = "normal_mode"
```

The theme is set to verify that `config.toml` is being successfully loaded. While my understanding is that the mechanism for the behavior of C-[ and Esc matching on Linux is a property of the terminal, it would be nice to match out-of-the-box for multiplatform support, or at least allow the explicit mapping.

### Environment

- Platform: Windows 10 Pro 21H1, 
- Helix version: `helix 0.4.1`
- Keyboard: Microsoft Natural Ergonomic Keyboard 4000 v1.0

The `helix.log` file is blank.

**Comment by superlou** [comment]
Not sure if it helps narrow down what's going on, but "A-[" maps via config.toml. It seems "C-]" and "C-." aren't recognized either, but "C-c" is. Something about Ctrl plus non-alphanumeric characters?

**Comment by WindSoilder** [comment]
Trying to investigate on it, and find that it's the issue on [crossterm](https://github.com/crossterm-rs/crossterm/issues/536)

**Comment by WindSoilder** [comment]
Have open a PR for this scenario https://github.com/crossterm-rs/crossterm/pull/619

Let's see how it goes... Hope that implementation is ok :-)

**Comment by WindSoilder** [comment]
> Your patch worked on my system to get crossterm to recognize the offending sequences

Glad that works :-)

> is there a way it can be applied at the helix level

Sadly I don't think so, because the relative [parse function](https://github.com/crossterm-rs/crossterm/blob/master/src/event/sys/windows/parse.rs#L57) is private, and it's hard to adjust outside crossterm.

>  it would be a shame to have to maintain a forked crossterm.

Agree with that, what if we just modify our `cargo.toml` locally?  Make the `crossterm` dependency temporary connect to other repo?  As example, we can modify [crossterm dependency](https://github.com/helix-editor/helix/blob/master/helix-term/Cargo.toml#L34) to something like this:
```toml
crossterm = { version = "0.22", features = ["event-stream"], git = "https://github.com/WindSoilder/crossterm" }
```

And we can revert it back as soon as that original issue is solved.

Here is the reference: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-git-repositories

**Comment by superlou** [comment]
Seeing the mention of a switch to termwiz in https://github.com/helix-editor/helix/issues/1472, is there any chance termwiz handles this? It seems like crossterm isn't accepting PRs very quickly.

**Comment by superlou** [comment]
It looks like @WindSoilder 's PR has landed in [Crossterm 0.23](https://github.com/crossterm-rs/crossterm/releases/tag/0.23)!
